lua/nvim-lsp-ts-utils/client.lua

```diff
@@ -36,6 +39,36 @@ local edit_handler = function(_, _, workspace_edit)
     return { applied = status, failureReason = result }
 end
 
+local diagnostics_handler = function(err, method, result, client_id, buf_nr, config)
+    local forbidden_severities = o.get().filter_out_diagnostics_by_severity
+    local forbidden_codes = o.get().filter_out_diagnostics_by_code
+
+    if #forbidden_severities > 0 or #forbidden_codes > 0 then
+        local filtered_diagnostics = vim.tbl_filter(function(diagnostic)
+            -- Only filter out Typescript LS diagnostics
+            if diagnostic.source ~= "typescript" then
+                return true
+            end
+
+            -- Filter out diagnostics with forbidden severity
+            if vim.tbl_contains(forbidden_severities, diagnostic.severity) then
+                return false
+            end
+
+            -- Filter out diagnostics with forbidden code
+            if vim.tbl_contains(forbidden_codes, diagnostic.code) then
+                return false
+            end
+
+            return true
+        end, result.diagnostics)
+
+        result.diagnostics = filtered_diagnostics
+    end
+
+    return vim.lsp.diagnostic.on_publish_diagnostics(err, method, result, client_id, buf_nr, config)
```

Comment:
We should use `lsp.handlers["textDocument/publishDiagnostics"]` to respect options users have set with `vim.lsp.with` (e.g. disabling virtual text).